package/generateScss.mjs

```diff
@@ -58,20 +59,89 @@ const generateSCSSModule = async (moduleName, importObj) => {
       }
       generatedScss += `${key}: ${value};\n`;
     });
-    
+  
+  // media.scss
   } else if (moduleName.toLowerCase() === 'media') {
     Object.keys(importObj).forEach((queryName) => {
       const processedQuery = customMediaHelper.process(queryName);
       queryName = queryName.replace('--', '$');
       generatedScss += `${queryName}: '${processedQuery}';\n`;
     });
+  
+  // shadows.scss
+  } else if (moduleName.toLowerCase() === 'shadows') {
+    
+    let mapKeysValues = '';
+    let lightColor = '';
+    let lightStrength = '';
+    let darkColor = '';
+    let darkStrength = '';
+    const entries = Object.entries(importObj);
+    
+    for (let index = 0; index < entries.length; index++) {
+      let [key, value] = entries[index];
+      
+      if (key == '--shadow-color') {
+        lightColor = value;
+      }    
+      if (key == '--shadow-strength') {
+        lightStrength = value;
+      }
+      if (key.includes('color-@media:dark')) {
+        darkColor = value;
+      }    
+      if (key.includes('strength-@media:dark')) {
+        darkStrength = value;
+      }
+      if (key == '--shadow-color' || key == '--shadow-strength' || key.includes('@')) {
+        continue; // skip light and dark for the other loops
+      } 
+
+      key = key.replace('--shadow-', '');
+      if (key.includes('--inner-shadow-')) {
+        key = key.replace('--inner-shadow-', '\'inner-');
+        key = key.replace(/$/, '\'');
+      }
+     
+      value = value.replace(/var\(--(.*?)\)/g, '$$$1');
+      value = value.replace(/hsl/g, 'Hsl')
+      mapKeysValues += `${key}: (${value})`;
+      
+      if (index < entries.length - 1) {
+        mapKeysValues += ',\n '; // Add comma and new line for all entries except the last one
+      }
+    };
     
+    generatedScss += `@use 'sass:map';
+
+$-shadow-color: null;
+$-shadow-strength: null;
+
+@mixin apply-shadows($shadow-color: null, $shadow-strength: null) {
```

Comment:
the main reason i'm hesitant to add this mixin: it can cause global side-effects. imagine you call this mixin somewhere and forget about it, then later you (or worse, someone else) uses `shadow()` and gets confused by it

compared to only having a simple `shadow()` which is pure/stateless.

for additional convenience (to avoid repetition), the user can always create a wrapper function
```scss
@function custom-shadow($level) {
  @return op.shadow($level, $shadow-strength: var(-foo));
}
```